src/rgw/driver/rados/rgw_zone.h

```diff
@@ -525,47 +525,56 @@ WRITE_CLASS_ENCODER(RGWPeriodConfig)
 class RGWRealm;
 class RGWPeriod;
 
-class RGWRealm : public RGWSystemMetaObj
+class RGWRealm
```

Comment:
love it!

it looks like `RGWRealm::set_current_period()` can also be removed in favor of `rgw::realm_set_current_period()`